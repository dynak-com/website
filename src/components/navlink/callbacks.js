export function onResize(navLinkRef) {
  if (window.innerWidth >= 992) {
    navLinkRef.current.dataset.toggle = '';
  } else {
    navLinkRef.current.dataset.toggle = 'collapse';
  }
}
