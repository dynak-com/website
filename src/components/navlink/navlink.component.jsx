import React from 'react';
import useComponent from './hook.js';
import styles from './styles.module.css';
import { NavLink as Container } from 'react-router-dom';

const CustomNavLink = ({ text, address, target }) => {
  const { navLinkRef } = useComponent();
  return (
    <Container exact to={ address } className={ styles.customNavlink } activeClassName={ styles.activated }>
      <span
        ref={navLinkRef}
        className={`nav-link d-inline-block d-lg-block px-1 my-1 my-lg-0 mx-lg-3` }
        data-target={`#${target}`}
      >
        {text}
      </span>
    </Container>
  );
};

export default CustomNavLink;
