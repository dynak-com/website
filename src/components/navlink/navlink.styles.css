.navbar-brand > #dot {
    color: #3eafad;
  font-size: 3.0rem;
}
.navbar-brand > a {
  transform: none;
}

#custom-navlink {
  transition: 0.3s;
}
a:hover > span#custom-navlink {
  background-color: #3eafad;
  text-decoration: none;
  transform: skew(-12deg);
  text-decoration: none;
}

a:link.active > span#custom-navlink {
  background-color: #3eafad;
  text-decoration: none;
  transform: skew(-12deg);
  text-decoration: none;
}

@media screen and (max-width: 992px) {
  #custom-navlink:link.active > span {
    transform: none;
  }
}
