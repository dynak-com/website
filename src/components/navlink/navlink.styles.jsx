import React from 'react';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const StyledNavLink = styled(NavLink)`
  h1 > span {
    font-size: 2.4rem;
  }
  span {
    transition: 0.3s;
  }
  &:focus,
  &:hover,
  &:visited,
  &:link {
    text-decoration: none;
    &.active > span {
      background-color: #3eafad;
      transform: skew(-12deg);
      @media screen and (max-width: 992px) {
        transform: none;
      }
    }
  }
`;

export default props => <StyledNavLink {...props} />;
