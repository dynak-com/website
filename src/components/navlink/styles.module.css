.navlink h1 > span {
  font-size: 2.4rem;
}

.navlink span {
  transition: 0.3s;
}

.navlink:focus,
.navlink:hover,
.navlink:visited,
.navlink:link {
  text-decoration: none;
}

.navlink:link.active > span {
  background-color: #3eafad;
  transform: skew(-12deg);
}

@media screen and (max-width: 992px) {
  .navlink:link.active > span {
    transform: none;
  }
}
