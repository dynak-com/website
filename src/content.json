[
  {
    "lang": "pl",
    "nav": [
      {
        "text": "nav.home",
        "address": "/"
      },
      {
        "text": "nav.about",
        "address": "/about"
      },
      {
        "text": "nav.projects",
        "address": "/projects"
      },
      {
        "text": "nav.contact",
        "address": "/contact"
      }
    ],
    "home": {
      "header": "home.header",
      "posts": [
        {
          "id": 1,
          "text": "home.posts.paragraph1",
          "headerText": "home.posts.headerText1"
        },
        {
          "id": 2,
          "text": "home.posts.paragraph2",
          "headerText": "home.posts.headerText2"
        },
        {
          "id": 3,
          "text": "home.posts.paragraph3",
          "headerText": "home.posts.headerText3"
        }
      ]
    },
    "about": {
      "header": "about.header",
      "technologies": {
        "title": "about.technologies.title",
        "content": "about.technologies.content"
      },
      "team": {
        "title": "about.team.title",
        "teamData": [
          {
            "name": "Emilia Dynak",
            "picture": "https://secure.gravatar.com/avatar/d34a1b21822840695b16113034ba740b?s=180&d=identicon",
            "position": "CEO and co-owner",
            "linkedInUrl": "https://www.linkedin.com/in/emilia-dynak-b0a3a9140/",
            "githubUrl": "https://github.com/edynak",
            "username": "edynak"
          },
          {
            "name": "Cezary Dynak",
            "picture": "https://secure.gravatar.com/avatar/47fbd06257cedcf9594bd9144f948feb?s=180&d=identicon",
            "position": "CTO and co-owner",
            "linkedInUrl": "https://www.linkedin.com/in/cdynak/",
            "githubUrl": "https://github.com/cdynak",
            "username": "cdynak"
          },
          {
            "name": "Natalia Borzyszkowska",
            "picture": "https://secure.gravatar.com/avatar/2cba91020f68596ad56766847694b292?s=180&d=identicon",
            "position": "CFO",
            "linkedInUrl": "https://www.linkedin.com/in/natalia-borzyszkowska-b7814218b/",
            "githubUrl": "https://github.com/login",
            "username": "21_smakow_natki"
          },
          {
            "name": "Aleksandra Borodziuk",
            "picture": "https://secure.gravatar.com/avatar/865d01afdfe703b4ef137693198b8500?s=180&d=identicon",
            "position": "CMO",
            "linkedInUrl": "https://www.linkedin.com/in/aleksandra-borodziuk/",
            "githubUrl": "https://github.com/login",
            "username": "Forestgirl25"
          },
          {
            "name": "Dominika Konefał",
            "picture": "https://secure.gravatar.com/avatar/ec094fb729d8e3509391628d8ba00dd8?s=180&d=identicon",
            "position": "Graphic Designer",
            "linkedInUrl": "https://www.linkedin.com/in/dominika-konefa%C5%82-028920185/",
            "githubUrl": "https://github.com/login",
            "username": "Deloi"
          },
          {
            "name": "Mikołaj Tumalewicz",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/2270247/avatar.png?width=180",
            "position": "DevOps/SysOps Engineer",
            "linkedInUrl": "https://www.linkedin.com/in/mtumalewicz/",
            "githubUrl": "https://github.com/mtumalewicz",
            "username": "mtumalewicz"
          },
          {
            "name": "Marcin Banik",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/2342806/avatar.png?s=180",
            "position": "Front-end Developer",
            "linkedInUrl": "https://www.linkedin.com/in/marcin-banik-947954142/",
            "githubUrl": "https://github.com/m-banik",
            "username": "mbanik"
          },
          {
            "name": "Bartosz Marciniec",
            "picture": "https://gitlab.com/uploads/-/system/user/avatar/4684288/avatar.png?s=180",
            "position": "Back-end Developer",
            "linkedInUrl": "https://www.linkedin.com/in/bartosz-marciniec-b0917517a/",
            "githubUrl": "https://github.com/hendas",
            "username": "hendas"
          },
          {
            "name": "Cristina Volk",
            "picture": "https://assets.gitlab-static.net/uploads/-/system/user/avatar/5111565/avatar.png?s=180",
            "position": "Web Developer",
            "linkedInUrl": "https://www.linkedin.com/in/kristina-volk-a9829517b/",
            "githubUrl": "https://github.com/CristinaVolk",
            "username": "CristinaVolk"
          }
        ]
      },
      "values": {
        "title": "about.values.title",
        "content": [
          "about.values.content.paragraph1",
          "about.values.content.paragraph2",
          "about.values.content.paragraph3"
        ]
      }
    },
    "projects": {
      "header": "projects.header",
      "posts": [
        {
          "id": 1,
          "text": "projects.posts.text1",
          "headerText": "dynak.io"
        },
        {
          "id": 2,
          "text": "projects.posts.text2",
          "headerText": "warehouse-app"
        },
        {
          "id": 3,
          "text": "projects.posts.text3",
          "headerText": "projects.posts.headerText3"
        }
      ]
    },
    "contact": {
      "header": "contact.header",
      "socials": "contact.socials",
      "form": {
        "subject": "contact.form.subject",
        "email": "contact.form.email",
        "message": "contact.form.message",
        "contactAgreement": "contact.form.contactAgreement",
        "privacyPolicyAgreement": "contact.form.privacyPolicyAgreement",
        "policyPrivacy": "contact.form.policyPrivacy",
        "send": "contact.form.send"
      }
    },
    "modal": {
      "modal_ok": {
        "modal_id": "modal_ok",
        "error_message": false,
        "thanks": "modal_ok.thanks",
        "sent": "modal_ok.sent",
        "close": "modal_ok.close"
      },
      "modal_error": {
        "modal_id": "modal_error",
        "error_message": true,
        "thanks": "modal_error.thanks",
        "sent": "modal_error.sent",
        "close": "modal_error.close"
      },
      "modal_privacy_policy": {
        "modal_id": "modal_privacy_policy",
        "error_message": null,
        "thanks": "modal_privacy_policy.thanks",
        "sent": "modal_privacy_policy.sent",
        "close": "modal_privacy_policy.close",
        "rodoContent": {
          "header": "rodoContent.header",
          "sentence1": "rodoContent.sentence1",
          "website": "rodoContent.website",
          "sentence2": "rodoContent.sentence2",
          "generalProvisions": "rodoContent.generalProvisions",
          "text1": "rodoContent.text1",
          "text2": "rodoContent.text2",
          "text3": "rodoContent.text3",
          "definitions": "rodoContent.definitions",
          "administrator": "rodoContent.administrator",
          "user": "rodoContent.user",
          "personalData": "rodoContent.personalData",
          "paragraph1": "rodoContent.paragraph1",
          "linkedIn": "rodoContent.linkedIn",
          "paragraph2": "rodoContent.paragraph2",
          "paragraph3": "rodoContent.paragraph3",
          "paragraph4": "rodoContent.paragraph4",
          "paragraph5": "rodoContent.paragraph5",
          "paragraph6": "rodoContent.paragraph6",
          "paragraph7": "rodoContent.paragraph7",
          "paragraph8": "rodoContent.paragraph8",
          "webLinkedIn": "rodoContent.webLinkedIn",
          "github": "rodoContent.github",
          "gitlab": "rodoContent.gitlab",
          "paragraph9": "rodoContent.paragraph9",
          "paragraph10": "rodoContent.paragraph10",
          "paragraph11": "rodoContent.paragraph11",
          "paragraph12": "rodoContent.paragraph12",
          "paragraph13": "rodoContent.paragraph13",
          "paragraph14": "rodoContent.paragraph14",
          "paragraph15": "rodoContent.paragraph15",
          "paragraph16": "rodoContent.paragraph16",
          "paragraph17": "rodoContent.paragraph17",
          "paragraph18": "rodoContent.paragraph18",
          "paragraph19": "rodoContent.paragraph19",
          "paragraph20": "rodoContent.paragraph20",
          "paragraph21": "rodoContent.paragraph21",
          "paragraph22": "rodoContent.paragraph22",
          "paragraph23": "rodoContent.paragraph23",
          "forms": "rodoContent.forms",
          "paragraph24": "rodoContent.paragraph24",
          "paragraph25": "rodoContent.paragraph25",
          "technologies": "rodoContent.technologies",
          "Administrator": "rodoContent.Administrator",
          "not": "rodoContent.not",
          "paragraph26": "rodoContent.paragraph26",
          "cookies": "rodoContent.cookies",
          "part1": "rodoContent.part1",
          "part2": "rodoContent.part2",
          "part3": "rodoContent.part3",
          "part4": "rodoContent.part4",
          "created": "rodoContent.created"
        }
      }
    },
    "error": "error",
    "footer": {
      "rodo": "footer.rodo",
      "dynak": "footer.dynak"
    }
  }
]