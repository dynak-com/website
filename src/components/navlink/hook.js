import { useEffect, useRef } from 'react';
import { onResize } from './callbacks';

const useComponent = () => {
  const navLinkRef = useRef();
  useEffect(() => {
    onResize(navLinkRef);
    window.addEventListener('resize', () => onResize(navLinkRef));
    return () =>
      window.removeEventListener('resize', () => onResize(navLinkRef));
  }, []);
  return {
    navLinkRef
  };
};

export default useComponent;
