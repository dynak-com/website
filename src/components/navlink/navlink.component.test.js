import React from 'react';
import { shallow } from 'enzyme';
import { CustomNavLink } from './navlink.component';
import content from '../../content.json';

describe('CustomNavLink component', () => {
    const [contentData] = content;
    const { nav } = contentData;

    it('renders correctly according to Snapshot', () => {
        nav.map((item) => {
            const { text, address } = item;
            const target = 'headerNavbarMenu';
            const wrapper = shallow(<CustomNavLink text={text} address={address} target={target} />);
            return expect(wrapper.exists()).toBeTruthy();
        });
    });
});
