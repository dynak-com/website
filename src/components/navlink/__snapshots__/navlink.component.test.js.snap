// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomNavLink component renders correctly according to Snapshot 1`] = `
"<NavLink exact={true} to=\\"/\\" className=\\"customNavlink\\" activeClassName=\\"activated\\">
  <span className=\\"nav-link d-inline-block d-lg-block px-1 my-1 my-lg-0 mx-lg-3\\" data-target=\\"#headerNavbarMenu\\">
    home
  </span>
</NavLink>"
`;

exports[`CustomNavLink component renders correctly according to Snapshot 2`] = `
"<NavLink exact={true} to=\\"/about\\" className=\\"customNavlink\\" activeClassName=\\"activated\\">
  <span className=\\"nav-link d-inline-block d-lg-block px-1 my-1 my-lg-0 mx-lg-3\\" data-target=\\"#headerNavbarMenu\\">
    o nas
  </span>
</NavLink>"
`;

exports[`CustomNavLink component renders correctly according to Snapshot 3`] = `
"<NavLink exact={true} to=\\"/projects\\" className=\\"customNavlink\\" activeClassName=\\"activated\\">
  <span className=\\"nav-link d-inline-block d-lg-block px-1 my-1 my-lg-0 mx-lg-3\\" data-target=\\"#headerNavbarMenu\\">
    projekty
  </span>
</NavLink>"
`;

exports[`CustomNavLink component renders correctly according to Snapshot 4`] = `
"<NavLink exact={true} to=\\"/contact\\" className=\\"customNavlink\\" activeClassName=\\"activated\\">
  <span className=\\"nav-link d-inline-block d-lg-block px-1 my-1 my-lg-0 mx-lg-3\\" data-target=\\"#headerNavbarMenu\\">
    kontakt
  </span>
</NavLink>"
`;
